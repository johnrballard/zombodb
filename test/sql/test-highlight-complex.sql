select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric", "david"], "last": "ridge", "id":{"low":0, "high":8}}}'::jsonb,
                            'comment_text.first = "david", comment_text.id.low = 0')
order by field_name, array_index, start_offset;


select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*", comment_text.id.high = 42')
order by field_name, array_index, start_offset;

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*" and comment_text.first:"joe", comment_text.id.high = 42')
order by field_name, array_index, start_offset;

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*" and comment_text.first:"david", comment_text.id.high = 42')
order by field_name, array_index, start_offset;

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*" with comment_text.first:"david", comment_text.id.high = 42')
order by field_name, array_index, start_offset;


