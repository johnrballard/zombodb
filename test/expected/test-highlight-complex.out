select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric", "david"], "last": "ridge", "id":{"low":0, "high":8}}}'::jsonb,
                            'comment_text.first = "david", comment_text.id.low = 0')
order by field_name, array_index, start_offset;
     field_name      | array_index | term  |      type      | position | start_offset | end_offset |        query_clause        
---------------------+-------------+-------+----------------+----------+--------------+------------+----------------------------
 comment_text.first  |           1 | david | <ALPHANUM>     |        1 |            0 |          5 | comment_text.first="david"
 comment_text.id.low |           0 | 0     | <UNSIGNED_INT> |        1 |            0 |          0 | comment_text.id.low="0"
(2 rows)

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*", comment_text.id.high = 42')
order by field_name, array_index, start_offset;
      field_name      | array_index | term  |      type      | position | start_offset | end_offset |         query_clause         
----------------------+-------------+-------+----------------+----------+--------------+------------+------------------------------
 comment_text.first   |           0 | eric  | <ALPHANUM>     |        1 |            0 |          4 | comment_text.first="eric r*"
 comment_text.first   |           0 | ridge | <ALPHANUM>     |        2 |            5 |         10 | comment_text.first="eric r*"
 comment_text.id.high |           0 | 42    | <UNSIGNED_INT> |        1 |            0 |          0 | comment_text.id.high="42"
(3 rows)

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*" and comment_text.first:"joe", comment_text.id.high = 42')
order by field_name, array_index, start_offset;
      field_name      | array_index | term |      type      | position | start_offset | end_offset |       query_clause        
----------------------+-------------+------+----------------+----------+--------------+------------+---------------------------
 comment_text.id.high |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | comment_text.id.high="42"
(1 row)

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*" and comment_text.first:"david", comment_text.id.high = 42')
order by field_name, array_index, start_offset;
      field_name      | array_index | term  |      type      | position | start_offset | end_offset |         query_clause         
----------------------+-------------+-------+----------------+----------+--------------+------------+------------------------------
 comment_text.first   |           0 | eric  | <ALPHANUM>     |        1 |            0 |          4 | comment_text.first="eric r*"
 comment_text.first   |           0 | ridge | <ALPHANUM>     |        2 |            5 |         10 | comment_text.first="eric r*"
 comment_text.first   |           1 | david | <ALPHANUM>     |        1 |            0 |          5 | comment_text.first:"david"
 comment_text.id.high |           0 | 42    | <UNSIGNED_INT> |        1 |            0 |          0 | comment_text.id.high="42"
(4 rows)

select *
from zdb.highlight_document('so_comments',
                            '{"comment_text": { "first": ["eric ridge", "david"], "last": "ridge", "id":{"low":0, "high":42}}}'::jsonb,
                            'comment_text.first = "eric r*" with comment_text.first:"david", comment_text.id.high = 42')
order by field_name, array_index, start_offset;
      field_name      | array_index | term |      type      | position | start_offset | end_offset |       query_clause        
----------------------+-------------+------+----------------+----------+--------------+------------+---------------------------
 comment_text.id.high |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | comment_text.id.high="42"
(1 row)

